source/internal.d.ts

```diff
@@ -317,3 +320,398 @@ IsPrimitive<Object>
 ```
 */
 export type IsPrimitive<T> = [T] extends [Primitive] ? true : false;
+
+/**
+Returns a boolean for whether A and B are both true.
+
+@example
+```
+And<true, true>
+//=> true
+
+And<true, false>
+//=> false
+```
+*/
+export type And<A extends boolean, B extends boolean> = [A, B][number] extends true
+	? true
+	: true extends [IsEqual<A, false>, IsEqual<B, false>][number]
+		? false
+		: boolean;
+
+/**
+Returns a boolean for either A or B is true.
+
+@example
+```
+Or<true, false>
+//=> true
+
+Or<false, false>
+//=> false
+```
+*/
+export type Or<A extends boolean, B extends boolean> = [A, B][number] extends false
+	? false
+	: true extends [IsEqual<A, true>, IsEqual<B, true>][number]
+		? true
+		: boolean;
+
+/**
+Returns a boolean for whether A is false.
+
+@example
+```
+Not<true>
+//=> false
+
+Not<false>
+//=> true
+```
+*/
+export type Not<A extends boolean> = A extends true
+	? false
+	: A extends false
+		? true
+		: boolean;
+
+/**
+Returns the maximum element in an array.
+
+Note:
+- float number is not supported.
+
+@example
+```
+Max<[1, 2, 5, 3]>
+//=> 5
+
+Max<[1, 2, 5, 3, 99, -1]>
+//=> 99
+```
+*/
+export type Max<A extends number[], Result extends number = NegativeInfinity> = A extends [infer F extends number, ...infer R extends number[]]
```

Comment:
```suggestion
export type ArrayMax<A extends number[], Result extends number = NegativeInfinity> = A extends [infer F extends number, ...infer R extends number[]]
```

Since this only applies to arrays.